tests/unit/CoreSeparatorTest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace WPGraphQL\ContentBlocks\Unit;
+
+final class CoreSeparatorTest extends PluginTestCase {
+	/**
+	 * The ID of the post created for the test.
+	 *
+	 * @var int
+	 */
+	public $post_id;
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->post_id = wp_insert_post(
+			[
+				'post_title'   => 'Post with Separator',
+				'post_content' => '',
+				'post_status'  => 'publish',
+			]
+		);
+
+		\WPGraphQL::clear_schema();
+	}
+
+	public function tearDown(): void {
+		parent::tearDown();
+
+		wp_delete_post( $this->post_id, true );
+
+		\WPGraphQL::clear_schema();
+	}
+
+	public function query(): string {
+		return '
+			fragment CoreSeparatorBlockFragment on CoreSeparator {
+				attributes {
+					align
+					anchor
+					backgroundColor
+					className
+					cssClassName
+					gradient
+					lock
+					# metadata
+					opacity
+					style
+				}
+			}
+
+			query Post( $id: ID! ) {
+				post(id: $id, idType: DATABASE_ID) {
+					databaseId
+					editorBlocks {
+						apiVersion
+						blockEditorCategoryName
+						clientId
+						cssClassNames
+						innerBlocks {
+							name
+						}
+						isDynamic
+						name
+						parentClientId
+						renderedHtml
+						...CoreSeparatorBlockFragment
+					}
+				}
+			}
+		';
+	}
+
+	/**
+	 * Test the retrieval of core/separator block fields and attributes.
+	 *
+	 * This test verifies that the block attributes for the core/separator block are
+	 * properly returned via the GraphQL query. The test ensures that the block content,
+	 * client ID, block name, and rendered HTML are correctly returned and that the attributes
+	 * (such as align, className, gradient, opacity, style and lock) match the expected values.
+	 *
+	 * @return void
+	 */
+	public function test_core_separator_fields_and_attributes() {
+		$block_content = '
```

Comment:
Use the naming patterns of the plugin you're contributing to.

```suggestion
	public function test_retrieve_core_table_attribute_fields(): void {
```